app/registries/foreman/setting_manager.rb

```diff
@@ -37,25 +37,32 @@ def storage
         SettingManager.settings
       end
 
+      def available_types
+        [:boolean, :integer, :float, :string, :text, :hash, :array]
+      end
+
       # Adds setting definition
       #
       # ===== Example
       #
       #   SettingManager.define(:puppet) do
       #     category(:cfgmgmt, N_('Configuration Management')) do
       #       setting(:use_cooler_puppet,
+      #               type: :boolean,
       #               default: true,
       #               description: N_('Use Puppet that goes to 11'),
       #               full_name: N_('Use shiny puppet'),
       #               encrypt: true)
```

Comment:
not directly related, but just noticed this should probably be
```suggestion
      #               encrypted: true)
```
(also in the plugin registry above)

Reply:
I'll do this separatelly, but good catch